public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -368,7 +401,7 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
       const target: InfluxQuery = {
         refId: 'metricFindQuery',
         datasource: this.getRef(),
-        query: this.templateSrv.replace(options.annotation.query ?? '', undefined, 'regex'),
+        query: this.templateSrv.replace(annotation.query ?? '', undefined, 'regex'),
```

Comment:
since we checked the `annotation.query` on line 393 it is ok to pass it directly.
```suggestion
        query: this.templateSrv.replace(annotation.query, undefined, 'regex'),
```

Reply:
Good catch!